File: 0007-reverse-integer/0007-reverse-integer.java
class Solution {
    public int reverse(int x) {
        int rev = 0;
        while (x != 0) {
            int remainder = x % 10;
            
            // Check for overflow/underflow
            if (rev > Integer.MAX_VALUE / 10 || (rev == Integer.MAX_VALUE / 10 && remainder > 7)) {
                return 0; // Overflow
            }
            if (rev < Integer.MIN_VALUE / 10 || (rev == Integer.MIN_VALUE / 10 && remainder < -8)) {
                return 0; // Underflow
            }
            
            rev = rev * 10 + remainder;
            x /= 10;
        }
        return rev;
    }
}

File: 0007-reverse-integer/0007-reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        res=0
        if x<0:
            res=int(str(x)[1:][::-1])*-1
        else:
            res=int(str(x)[::-1]) 
        if res>= -2**31 and res<=2**31-1:
            return res
        else:
            return 0

    
